os/drivers/lcd/lcd_dev.c

```diff
@@ -356,6 +399,19 @@ int lcddev_register(struct lcd_dev_s *dev)
 
 	priv->dev = dev;
 
+	sem_init(&flushing_sem, 0, 0);
+
+	g_wait = false;
+	empty = true;
+
+	lcd_kbuffer = (uint8_t *)kmm_malloc(CONFIG_LCD_XRES * CONFIG_LCD_YRES * sizeof(uint16_t));
```

Comment:
This seems to be in lcd_dev_s.
Please remove the global variable.

Reply:
I have added the variables in a separate structure.
We can't pass via lcd_dev_s to thread function because it's changing the address. I checked by implementing it.